### DISCUSSION TITLE
Il file specificato non esiste.

### DISCUSSION THREAD

**Discussion by khizar-wahab** [fattureincloud/api#290] [Q&A] [answered]
Hello I am trying to create expense with attachment. First i upload attachment using "createReceivedDocument" method and then send the returned token in payload to create expense but it gives this error "Il file specificato non esiste.". What could be the problem? Thanks.

**Comment by emajo** [comment, marked as answer]
Hi @khizar-wahab the problem you descibed happens when you pass to the createReceivedDocument method an `attachment_token` that does not exists.

Thanks,
Emanuele.

**Reply by khizar-wahab** [reply]
But when i upload attachment it does not give any error but instead returns the token. Does it returns token even when the attachment is not successfully created? if so then what could be the reason?
I am using this image path "https://lh3.googleusercontent.com/d/10TurfRP9jut_bS4uMd2ecZRjbATfH4zD?filename=data.jpg"

**Reply by valmoz** [reply]
Hi @khizar-wahab 
the methods to upload attachments are the following:

- [Upload Archive Document Attachment](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/archive/attachment)
- [Upload Issued Document Attachment](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/issued_documents/attachment)
- [Upload Received Document Attachment](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/received_documents/attachment)
- [Upload F24 Attachment](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/taxes/attachment)

You should use the method related to the kind of document you're trying to generate.
These methods will return something like this:
```json
{
"data": {
"attachment_token": "FAKE_TOKEN"
}}
```

You should take the _data.attachment_token_ param and pass it in the body of the Create method. Please, be sure to follow this procedure and to use the correct Upload Attachment method, based on the kind of document you're creating.

Thanks
Mauro

**Reply by valmoz** [reply]
Hi @khizar-wahab were you able to solve your issue?

Thanks
Mauro
